src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1213,7 +1214,19 @@ static bool addTriggeredSubscriptions_withCache
       LM_T(LmtMongo, ("adding subscription: '%s'", sub.toString().c_str()));
 
       long long throttling       = sub.hasField(CSUB_THROTTLING) ? getField(sub, CSUB_THROTTLING).numberLong() : -1;
-      long long lastNotification = sub.hasField(CSUB_LASTNOTIFICATION) ? getIntField(sub, CSUB_LASTNOTIFICATION) : -1;
+      long long lastNotification = -1;
+
+      if (sub.hasField(CSUB_LASTNOTIFICATION))
```

Comment:
Actually, any integer field could have this problem.

CASUB_LASTNOTIFICATION does not, it is read/written only in two places:

    newSub.append(CASUB_LASTNOTIFICATION, getIntField(sub, CASUB_LASTNOTIFICATION));
    BSONObj     update = BSON("$set" << BSON(CASUB_LASTNOTIFICATION << getCurrentTime()) << "$inc" << BSON(CASUB_COUNT << 1));

Now, as getCurrentTime() returns **int** we are good.

I would like to stress a little on the danger of using the << operator. It's not the first time I see this.
If getCurrentTime() changes to reutn a long, the **<< getCurrentTime()** would still work, no warning would be issued but a Long field would be created in mongo and later when reading that fiels (as Int), the broker crashes ...